rusoto/credential/src/instance_metadata.rs

```diff
@@ -122,11 +136,15 @@ impl ProvideAwsCredentials for InstanceMetadataProvider {
 fn get_role_name(
     client: &HttpClient,
     timeout: Duration,
+    port_for_testing: Option<String>,
```

Comment:
Or we can just take `&Option<String>` here?

Reply:
Yes, that's worth looking at, my only reason not to do that was when I tried I got a bunch of compiler errors and decided to go to bed instead. 😆